src/components/Card/Card.test.tsx

```diff
@@ -1,37 +1,97 @@
-import { render } from '@testing-library/react';
 import React from 'react';
+import { render } from '@testing-library/react';
 
 import { Card, CardProps } from './';
+import styles from './Card.module.scss';
 
-const DEFAULT_PROPS: CardProps = {
-  title: 'Title Text',
-  value: '123',
-  bottomText: 'Bottom Text'
+const DEFAULT_PROPS = {
+  title: 'Mock Text Title',
+  value: 'Mock Text Value',
+  bottomText: 'Mock Text Bottom'
 };
 
+const MOCK_SKELETON = 'Mock Skeleton';
+
 jest.mock('../Skeleton', () => ({
-  Skeleton: () => <span>Skeleton</span>
+  // Note: if you change this from <div>, make sure to update tests below
+  Skeleton: () => <div>{MOCK_SKELETON}</div>
 }));
 
-const renderComponent = (props?: CardProps) => render(<Card {...props} />);
+const renderComponent = (props?: Partial<CardProps>) => render(<Card {...DEFAULT_PROPS} {...props} />);
+
+test('should render correct nodes in the correct order', () => {
+  const { container } = renderComponent(DEFAULT_PROPS);
+
+  // Extract text and nodes
+  const children = container.firstChild.childNodes;
+  const asText = Array.from(children).map(c => c.textContent);
+  const asNodes = Array.from(children).map(c => c.nodeName);
+
+  // Assert correct nodes and correct order
+  const expectedNodes = ['label', 'b', 'span'];
+  const expectedText = [DEFAULT_PROPS.title, DEFAULT_PROPS.value, DEFAULT_PROPS.bottomText];
+  expect(
+    asText.toString() === expectedText.toString() &&
+      asNodes.map(n => n.toLowerCase()).toString() === expectedNodes.map(n => n.toLowerCase()).toString()
+  );
+});
+
+test('should apply class module names to correct elements', () => {
+  const { getByText } = renderComponent();
+
+  expect(getByText(DEFAULT_PROPS.title)).toHaveClass(styles.Title);
+  expect(getByText(DEFAULT_PROPS.value)).toHaveClass(styles.Value);
+  expect(getByText(DEFAULT_PROPS.bottomText)).toHaveClass(styles.BottomText);
+});
```

Comment:
Sweet - have handled most of these. Agreed on making smaller tests.

I'm finding it difficult to remove `DEFAULT_PROPS`. I'm using it here because, if I didn't, I'd have to repeatedly define the props per test. If I added a new required prop to `Card`, I'd have to go through each test and add it, which adds friction.

Would `const { ... } = renderComponent(DEFAULT_PROPS)` be an acceptable middle-ground? This is more explicit than handling them in the `renderComponent` function, and lets me reduce repetition.

Reply:
@colbr maybe my comment just wasn't clear enough. my suggestion still solves the problem you're referring to. you shouldn't be defining all of the props in every test. what i am referring to is that you *should* be explicitly providing the props that are important to each given test. if you think it makes it more explicit to provide the defaults in each test, then maybe go for something like this:
`const { ... } = renderComponent(withDefaults({ title: 'my title!' }));`

(i will also add that setting sane defaults for every test, which happens if you have a global setup, actually hinders the scope of what you're testing. using the bare minimum for defaults (nulls for objects, empty strings, zero, etc.) actually helps make sure that your component handles situations where data isn't fully formed yet.)

Reply:
@jtulloch ohhh ok I think I read this wrong initially. Just to be clear - you're cool with using `DEFAULT_PROPS` in `renderComponent`, and then overriding `DEFAULT_PROPS` with whatever is passed to `renderComponent`? e.g.

```js
// psuedo
const renderComponent  = (props) => {
    return render(<Component {...props} {...DEFAULT_PROPS} />)
}

// in test
const { ... } = renderComponent({ title: 'mock title' });
```

I think I would prefer to use `const { ... } = render(<Component title="mock title" {...DEFAULT_PROPS} />)` the below if you think it's an acceptable option.

> i will also add that setting sane defaults for every test...

Is this relevant to these tests? I'm not really sure what the benefit of setting minimum defaults would be if we were passing all relevant props per test. I'm only really using `DEFAULT_PROPS` to satisfy types. I would think the non-sane values like `''` and `null` would be handled in tests. If possible could you provide an example from this test suite? I'm not quite sure I'm following and I think a concrete example would help.

Although now that I think about it, we haven't tested `bottomText is undefined`.